### PR TITLE
Implement permission changes

### PR DESCRIPTION
**Pull request by Lulalaby**
Update Permissions to include newly documented Permission

# Summary

Implements the following permissions to `DSharpPlus.Permissions`:

- `SEND_MESSAGES_IN_THREADS`
- `CREATE_PUBLIC_THREADS` (Replaces `USE_PUBLIC_THREADS` )
- `CREATE_PRIVATE_THREADS` (Replaces `USE_PRIVATE_THREADS`)
- `USE_APPLICATION_COMMANDS` (Replaces `USE_SLASH_COMMANDS`)
- `START_EMBEDDED_ACTIVITIES`

# Details

Adds the following permissions to the `Permissions` enum.
- `SendMessagesInThreads`
- `CreatePublicThreads`
- `CreatePrivateThreads`
- `UseApplicationCommands`
- `StartEmbeddedActivities`

Marked the following permissions as obsolete:
- `UsePublicThreads`
- `UsePrivateThreads`
- `UseSlashCommands`

# Changes proposed

  - Add `Permissions.SendMessagesInThreads` with a value of `0x0000004000000000`
  - Add `Permissions.StartEmbeddedActivities` with a value of `0x0000008000000000`

  - Modify the `FULL_PERMS` and `All` values to the result of the old `All` value and all new values resulting to a new value of `1099511627775`

# Notes
https://github.com/discord/discord-api-docs/pull/3672
https://github.com/discord/discord-api-docs/pull/3614/files#diff-83c324034ba0b1753dde3487f095f5307f7036c3e53a25bacba1879733a6ce23
https://github.com/discord/discord-api-docs/pull/3805

### CODE REVIEW

**Comment by Neuheit** [comment]
> 
> 
> @Neuheit is the obsolete attribute placed right?

Correct.

**Comment by Lulalaby** [comment]
Btw why on hold?

**Comment by VelvetThePanda** [comment]
> Btw why on hold?

We don't merge until Discord merges

**Comment by Lulalaby** [comment]
Discord already implemented the new permissions.

**Comment by VelvetThePanda** [comment]
Very much appears to still be open to me image

**Comment by Lulalaby** [comment]
True but it's already in clients available.
Guilds can decide to switch already to the new permission system.

**Comment by VelvetThePanda** [comment]
It kind of isn't, actually. They're offering to switch so that it will be on by default when it does happen. see

**Comment by Giggitybyte** [comment]
Community guilds have the option to change to the new permission system early 
permission screenshot

**Comment by VelvetThePanda** [comment]
I'm aware, but given that it's not out for everyone and the fact that that the PR is still open means that we should probably still wait

**Comment by Lulalaby** [comment]
image

**Comment by Lulalaby** [comment]
image
https://github.com/discord/discord-api-docs/pull/3805

**Comment by Lulalaby** [comment]
@Neuheit the pr is merged

**Comment by IDoEverything** [comment]
Correct me if I'm wrong, but doesn't this break `HasPermission` and `ToPermissionString` in some way? You're adding `[Obsolete]` but it still counts as a member of the enum, so I would imagine it has unintended effects there. Could you test and see what they return?

File: DSharpPlus/Enums/Permission.cs
Comment (Neuheit) [inline DSharpPlus/Enums/Permission.cs:323]:
```suggestion
        /// Allows for creating private threads.
```

File: DSharpPlus/Enums/Permission.cs
Comment (Neuheit) [inline DSharpPlus/Enums/Permission.cs:317]:
```suggestion
        /// Allows for creating private threads.
```

**Comment by Lulalaby** [comment]
> 
> 
> Correct me if I'm wrong, but doesn't this break `HasPermission` and `ToPermissionString` in some way? You're adding `[Obsolete]` but it still counts as a member of the enum, so I would imagine it has unintended effects there. Could you test and see what they return?

Gonna look later.

**Review by Neuheit — APPROVED** [review]

**Comment by Lulalaby** [comment]
Nice, thanks!

**Pull request merged**
